middleware/jwt-session.js

```diff
@@ -51,6 +56,50 @@ async function validateUser(req, res, next) {
   }
 }
 
+/** validateUserHasRequiredRoles
+ * @param permittedRoles: an array of strings naming role required to
+ * validate on the JWT
+ * @returns function: the route handler function called by express
+ * example:
+ *   // server.js
+ *   app.post(
+ *    "/accounts/protected",
+ *    jwtSession.validateUserHasRequiredRoles(["admin", "security_admin"]),
+ *    accountsController.performActionOnlyPermittedToAdminRole
+ *   );
+ */
+function validateUserHasRequiredRoles(permittedRoles) {
+  if (!permittedRoles || permittedRoles.length < 1) {
+    permittedRoles = ["date_entry"];
+  }
+  return async function validateUserJwt(req, res, next) {
+    const jwtString = req.headers.authorization || req.cookies.jwt;
+    try {
+      // the payload object encoded on the JWT
+      const payload = await verify(jwtString);
+
+      // check that JWT subject is encoded with at least one of the requiredRoles
+      let isJWTRoleInAllowedRoles = permittedRoles.some((permittedRole) => {
```

Comment:
way this works is that each route can specify a list of 1 or more roles that must exist on the users' jwt in order to proceed. 

e.g 
```javascript
app.get('/my-route', validateUserHasRequiredRoles(['admin', 'coordinator']), controllers.myHandler)
```
so that any calling user with either `admin` or `coordinator` roles should be permitted to continue, otherwise they're returned a 401 auth error